Shopify/cli
# Fix x request id

### WHY are these changes introduced?

Fixes https://github.com/Shopify/internal-cli-foundations/issues/561

### WHAT is this pull request doing?

- Remove the `X-Request-Id` we used to send to the partners graphQL api (it was ignored and immediately replaced).
- Read the `x-request-id` header from the partners response
- Adjust logging 

### How to test your changes?

- Run any command with `--verbose`, you should see a `x-request-id` token
- search for that on our logging infrastructure
- it should appear straight away

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
